### DISCUSSION TITLE
SPOTIFY_REFRESH_TOKEN

### DISCUSSION THREAD

**Discussion by hirockstar** [j471n/j471n.in#19] [Q&A] [answered]
bro i'm not able to genrate refresh token is there any easy way to do it.
or you have any postment api??

**Comment by j471n** [comment, marked as answer]
I made a blog to get the Spotify credentials including `refresh_token` you can take a look at that [here](https://j471n.in/blogs/spotify-api-nextjs). It shows a step-by-step guide to get it.

**Reply by hirockstar** [reply]
i have check that bro already but im stuck on this https://j471n.in/blogs/spotify-api-nextjs#authorization-code

**Reply by j471n** [reply]
Ok then, what kind of error you are facing. or what's the response? what's your URL looks like?

P.S. You need to encode the URL after providing code, redirect uri, client id and scope.

There is no other way to get the refresh token as shown in the diagram 
![image](https://user-images.githubusercontent.com/55713505/217175829-dc124f9a-ebae-46ca-87e4-b6fe1027c1bb.png)

**Reply by j471n** [reply]
@therockstarind Is your problem fixed?

**Reply by hirockstar** [reply]
no bro 🥲

**Reply by j471n** [reply]
Could you show me what response you are getting?

**Reply by hirockstar** [reply]
i just got the Client id and secert key but i don't know how to start the next step.

**Reply by j471n** [reply]
Ok What scopes you are trying to get? such as I wanted `user-read-currently-playing` and `user-top-read`.  Also if you can just me your encoded url that would be great. 

Like this:
```bash
https%3A%2F%2Faccounts.spotify.com%2Fauthorize%3Fclient_id%3Dbda10cd71956489881825d97727de7e%26response_type%3Dcode%26redirect_uri%3Dhttp%3A%2F%2Flocalhost%3A3000%2Fcallback%26scope%3Duser-read-currently-playing+user-top-read
```

**Don't share it here**. You can mail me at `j471n.dev@gmail.com`.

**Reply by hirockstar** [reply]
thanks bro done
